.github/workflows/ci.yml

```diff
@@ -1,9 +1,6 @@
 name: ci
 
 on:
-  push:
-    branches:
-      - main
```

Comment:
I think we still want to run tests on merge to main, since there are cases where a PR with passing tests can be merged and break main, if it isn't rebased before merging. I prefer not to require that branches are always rebased before merging since it adds a lot of overhead to the flow. Perhaps we can conditionally run the codecov step?

Reply:
I think you can do something like

```
        if: github.ref != 'refs/heads/main'
```

Reply:
Arg sorry, yes yes definitely I was going too fast on this one.

Reply:
> I think you can do something like
> 
> ```
>         if: github.ref != 'refs/heads/main'
> ```

Yes I think it's better that `event_name` to specifically ignore the main.
Thank you sensei.